RdlMigration/ConvertRDL.cs

```diff
@@ -560,6 +577,50 @@ private XElement CreateDataSourceNode(XDocument doc, DataSourceDefinition curren
             return currNode;
         }
 
+        private void DiscoverSubreports(XDocument doc)
+        {
+            var subreports = doc.Descendants(doc.Root.Name.Namespace + "Subreport");
+            var subreportPaths = new List<string>();
+            foreach (XElement subreport in subreports)
+            {
+                var subreportName = subreport.Elements().First().Value;
+                var subreportPath = Path.Combine(rootFolder, subreportName);
+                subreportPath = Path.GetFullPath(subreportPath).Replace("\\", "/");
+                subreportPath = subreportPath.Substring(subreportPath.IndexOf('/'));
+                subreportName = Path.GetFileName(subreportPath);
+                
+                try
+                {
+                    if (rdlFileIO.IsFolder(subreportPath))
+                    {
+                        Trace($"SUBREPORT FAIL : {subreportPath} Subreport does not exist");
+                        continue;
+                    }
+                }
+                catch (Exception)
+                {
+                    Trace($"SUBREPORT FAIL : {subreportPath} Subreport does not exist");
+                    continue;
+                }
+
+                if (!reportNameMap.ContainsKey(subreportName))
```

Comment:
>if (!reportNameMap.ContainsKey(subreportName)) [](start = 16, length = 46)

this will cause a race condition between ContainsKey and TryAdd. That is why tryAdd returns true/false you should just do if (reportNameMap.TryAdd(subreportName) { do work } instead

Reply:
will do.